WIP: Add SFA preprocessing

jraddaoui opened the pull request.

Commit:
Add SFA preprocessing

DanielCosme committed:
wip

jraddaoui committed:
Compress testdata SIP and move it to sampledata

jraddaoui committed:
Fix package extraction

jraddaoui committed:
Install python, pip and lxml in enduro-a3m-worker

Commit:
Preprocessing works

DanielCosme committed:
Clean up files

DanielCosme commented:
Currently using a3m. Failing at aip creation, not sure why. Bagit validation is commented out. Need to test if `am` will create aips. As far as I can tell the `pre-processing` step is done.

DanielCosme committed:
Update am processing workflow test

djjuhasz committed:
Zip transfer before uploading it to AM

jraddaoui commented on Dockerfile:
```diff
@@ -66,8 +66,34 @@ COPY --from=build-enduro-am-worker --link /src/enduro.toml /home/enduro/.config/
+RUN pip3 install --no-cache --upgrade pip lxml bagit==v1.8.1
 COPY --from=build-enduro-a3m-worker --link /out/enduro-a3m-worker /home/enduro/bin/enduro-a3m-worker
 COPY --from=build-enduro-a3m-worker --link /src/enduro.toml /home/enduro/.config/enduro.toml
+# SFA metadata schema for validation.
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/xsdval.py xsdval.py
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/repackage_sip.py repackage_sip.py
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/ablieferung.xsd ablieferung.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/archivischeNotiz.xsd archivischeNotiz.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/archivischerVorgang.xsd archivischerVorgang.xsd 
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/arelda.xsd arelda.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/base.xsd base.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/datei.xsd datei.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/dokument.xsd dokument.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/dossier.xsd dossier.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/ordner.xsd ordner.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/ordnungssystem.xsd ordnungssystem.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/ordnungssystemposition.xsd ordnungssystemposition.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/paket.xsd paket.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/provenienz.xsd provenienz.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/zusatzDaten.xsd zusatzDaten.xsd
+COPY --from=build-enduro-a3m-worker --link /src/hack/sampledata/xsd/bagit.txt bagit.txt
```
Why not copying the entire folder?

DanielCosme replied:
Because I need to files to be at the root of the container file system because of the lxml handling of relative and absolute paths that are passed into the python script. Is there a docker one liner that does this without creating a folder and leaving this files at the root?

jraddaoui replied:
I'd try to encapsulate the python script and related files in its own directory, but if you still want it on root, this should work:

```
COPY --from=build-enduro-am-worker --link /src/hack/sampledata/xsd /
```

I think that won't overwrite the entire root folder, but if it does, maybe:

```
COPY --from=build-enduro-am-worker --link /src/hack/sampledata/xsd/* /
```

DanielCosme replied:
It works! Thank you very much for the suggestion. I actually tried a version of this and failed, that was why I ended up with the long list.

b2d11356f607ddb31a54bf4917c7126a8578d7dd

DanielCosme committed:
Fix am pre-processing

DanielCosme committed:
wip

djjuhasz committed:
Fix creation and upload of zip file

- Don't include parent dirs when creating a zip file from a dir
- Use the zip file name as the remote file name in the upload transfer
  activity
- Fix a bug in the zip activity that set the destination file name to
  an empty string when `DestPath` was set

DanielCosme commented:
@scollazo For the deployment
- `bagit.py` dependency no longer required.
- The file `hack/sampledata/xsd/bagit.txt` has been deleted form the repo (no longer needed)
- The file `hack/sampledata/xsd/repackage_sip.py` has been deleted form the repo (no longer needed)

DanielCosme committed:
Create bag-info.txt when repackaging SiP

DanielCosme committed:
Send failed packages to "Failed" minIO bucket

DanielCosme committed:
Send failed SIPs to a different bucket.

jraddaoui commented:
Moved to #4, which is based on the main branch.

jraddaoui commented:
I kept the branch for now in case it's needed for deployments.
